### PR TITLE
feat: seperate node and browser loggers

### PR DESCRIPTION
**Pull request by zachferland**
~~will revisit tmrrw, some cleanup/types, and had some trouble testing locally and getting dependencies to resolve properly~~

likely better long term solutions after, but can be revisited at same time as we work on core working in browser again, there may be other options for libraries as well, but wanted to minimize any changes there since we have a lot built on current log formats.

### CODE REVIEW

**Comment by zachferland** [comment]
This and https://github.com/StableLib/stablelib/pull/30 will allow http-client to be bundled by default (tested webpack5, probably helps with other as well, or at least optimizes those by removing polyfills not needed) 

This is bit of a bandaid that minimizes making changes anywhere else, but we should trend towards organizing more like described here https://github.com/ceramicnetwork/js-ceramic/issues/1163 when we revisit/choose to prioritize getting ceramic core to work/bundle in browser

File: packages/common/src/logger-base.ts [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+  write: (message: string) => void
+}
+
+/**
+ * Logs to the console based on log level
+ */
+export class DiagnosticsLoggerBase {
+  public readonly logLevel: LogLevel;
+  protected logger: any;
+  protected readonly fileLogger: WriteableStream;
```
Comment (stbrody) [inline packages/common/src/logger-base.ts:26]: Seems like this should be moved out of the base class and into the `DiagnosticsLogger` that extends this and actually uses it?

File: packages/common/src/logger-base.ts [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+}
+
+/**
+//  * Global Logger factory
+//  */
+export class LoggerProviderBase {
+  public readonly config: LoggerConfig
+
+  protected readonly _diagnosticLogger: DiagnosticsLoggerBase
+  protected readonly _fileLoggerFactory: FileLoggerFactory
```
Comment (stbrody) [inline packages/common/src/logger-base.ts:160]: Ditto, seems like this doesn't belong in the base class if it's not used here anywhere

**Review by stbrody — APPROVED** [review]
LGTM mod question about moving file loggers out of base classes and into derived classes

**Comment by zachferland** [comment]
tried changing, but then remembered why i did that, otherwise the extended class types would not match the base class types which are then used everywhere else. (although not really sure why it complains when protected, public interfaces the same). The better implementation again would be complete separation and likely shared logger interfaces that they both implement instead.

**Comment by stbrody** [comment]
> tried changing, but then remembered why i did that, otherwise the extended class types would not match the base class types which are then used everywhere else. (although not really sure why it complains when protected, public interfaces the same). The better implementation again would be complete separation and likely shared logger interfaces that they both implement instead.

Ah I see. `LoggerProviderBase` would need to be called `LoggerProvider`, but then you'd need a factory for getting LoggerProvider instances, which defeats the whole purpose of the provider.  Okay, this is fine for now then until we can do the better fix

**Review by stbrody — APPROVED** [review]

**Pull request merged**